packages/builder/src/package.ts

```diff
@@ -15,6 +15,70 @@ export type CopyPackageOpts = {
   recursive?: boolean;
 };
 
+export interface CannonPackageReference {
+  name: string;
+  version: string;
+  preset: string;
+  formatted(): string;
+}
+
+/**
+ * Used to format any reference to a cannon package and split it into it's core parts
+ */
+export class PackageReference implements CannonPackageReference {
```

Comment:
@dbeal-eth I added this package reference class and added it to provision and import steps for parsing out name, version and preset from package sources.